src/debug-adapter/debugSessionProxy.ts

```diff
@@ -75,11 +75,16 @@ export class DebugSessionProxy {
   }
 
   async getAllCurrentVariables(): Promise<Variable[]> {
-    const scopes = await this.getAllCurrentScopes();
-    const reply = await Promise.all(
-      scopes.filter((s) => !s.expensive && s.name !== 'Global').map((s) => this.getVariables(s.variablesReference))
-    );
+    try {
+      const scopes = await this.getAllCurrentScopes();
+      const reply = await Promise.all(
+        scopes.filter((s) => !s.expensive && s.name !== 'Global').map((s) => this.getVariables(s.variablesReference))
+      );
 
-    return reply.reduce((p, c) => p.concat(c));
+      return reply.reduce((p, c) => p.concat(c));
+    } catch (error) {
+      console.error(error);
+      return [];
+    }
```

Comment:
* Exceptions are handled here based on assumptions about the (only) caller, which is the `onDidSendMessage` method.
* Is exception handling necessary at all (errors are logged silently by VS code anyway)?

Reply:
The changes to the getAllCurrentVariables function were made to make the exception handling more consistent over the whole class. It would be possible to remove the exception handling from getAllCurrentVariables but I think we should discuss how the exception handling is done in general in the debugSessionProxy class

Reply:
Good point. Approved.